Add example in C and Swift on macOS (#14)
mptcp-apps/mptcp-hello: objective-c/README.md

# Using Multipath TCP in Objective-C on iOS

> 

Reviewer: can you add a note that this is specific to iOS, and it is not supported on <how do you call the OS for the laptops/desktops? iOS is based on macOS, MacOSX?>

Author: Seems like it is OSX indeed, I'll add this note

# Using Multipath TCP in Objective-C on iOS

> To use Multipath TCP in Objective-C, you simply need to change a bit the configuration given to `NSURLSession`, by selecting a Multipath TCP mode. The list of available modes is given in [the official documentation](https://developer.apple.com/documentation/foundation/nsurlsessionmultipathservicetype). Note however that enabling MPTCP on `NSURLSession` can only be done on iOS, it isn't supported yet on OSX. Here is an example:

Reviewer: ```suggestion
To use Multipath TCP in Objective-C, you simply need to change a bit the configuration given to `NSURLSession`, by selecting a Multipath TCP mode. The list of available modes is given in [the official documentation](https://developer.apple.com/documentation/foundation/nsurlsessionmultipathservicetype).

Currently, enabling MPTCP on `NSURLSession` can only be done on **iOS**, **not OSX**.

Here is an example:
```

NSURLSession *session = [NSURLSession sessionWithConfiguration:conf];
```

In this repository, you'll also find categories that extend the `NSURLSession` and `NSURLSessionConfiguration`, which allows having a single shared instance, called `sharedMPTCPSession`, for `NSURLSession` with Multipath TCP enabled in handover mode.

> Another remark: only client side MPTCP is supported currently.

Reviewer: ```suggestion
Another remark: currently, only the client side is supported.
```
